packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/TemplateBasedProjectCreator.java

```diff
@@ -0,0 +1,77 @@
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.project.services;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IEditingContextPersistenceService;
+import org.eclipse.sirius.components.core.api.IEditingContextSearchService;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.web.application.project.dto.CreateProjectFromTemplateInput;
+import org.eclipse.sirius.web.application.project.dto.CreateProjectFromTemplateSuccessPayload;
+import org.eclipse.sirius.web.application.project.services.api.IProjectMapper;
+import org.eclipse.sirius.web.application.project.services.api.IProjectTemplateInitializer;
+import org.eclipse.sirius.web.application.project.services.api.ITemplateBasedProjectCreator;
+import org.eclipse.sirius.web.domain.boundedcontexts.project.services.api.IProjectSearchService;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.support.TransactionTemplate;
+
+/**
+ * Used to create project from templates.
+ *
+ * @author sbegaudeau
+ */
+@Service
+public class TemplateBasedProjectCreator implements ITemplateBasedProjectCreator {
+
+    private final IProjectSearchService projectSearchService;
+
+    private final IProjectMapper projectMapper;
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final IEditingContextPersistenceService editingContextPersistenceService;
+
+    private final TransactionTemplate transactionTemplate;
+
+    private final ApplicationEventPublisher applicationEventPublisher;
+
+    public TemplateBasedProjectCreator(IProjectSearchService projectSearchService, IProjectMapper projectMapper, IEditingContextSearchService editingContextSearchService, IEditingContextPersistenceService editingContextPersistenceService, TransactionTemplate transactionTemplate, ApplicationEventPublisher applicationEventPublisher) {
+        this.projectSearchService = Objects.requireNonNull(projectSearchService);
+        this.projectMapper = Objects.requireNonNull(projectMapper);
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.editingContextPersistenceService = Objects.requireNonNull(editingContextPersistenceService);
+        this.transactionTemplate = Objects.requireNonNull(transactionTemplate);
+        this.applicationEventPublisher = Objects.requireNonNull(applicationEventPublisher);
+    }
+
+    @Override
+    @Transactional
+    public IPayload createProjectFromTemplate(CreateProjectFromTemplateInput input, UUID projectId, IProjectTemplateInitializer initializer) {
```

Comment:
Not sure about the naming here. `ITemplateBasedProjectCreator.createProjectFromTemplate`, but the project has actually already been created when this is called. This is more a project content initializer or something like that.
It's also very close to an editing context action in what it can do; it's just invoked at a different time (right after the project is created) and from a different UI mechanism (from buttons in the projects page and not exposed once the project is created/opened).

Reply:
I was not fan of the name either, I'll change it.